lib/chef_zero/endpoints/principal_endpoint.rb

```diff
@@ -30,13 +30,22 @@ def get(request)
           end
         end
         if json
-          json_response(200, {
+          principal_data = {
             'name' => name,
             'type' => type,
             'public_key' => FFI_Yajl::Parser.parse(json)['public_key'] || PUBLIC_KEY,
             'authz_id' => '0'*32,
             'org_member' => org_member
-          })
+          }
+
+          response_data =
+            if request.api_v0?
+              principal_data
+            else
+              { "principals" => [ principal_data ] }
```

Comment:
Yeah, that's correct, according to my reading of the code here: https://github.com/chef/chef-server/blob/218579549024b26b3b97c9958503ff3651e3fb16/src/oc_erchef/apps/oc_chef_wm/src/chef_wm_named_principal.erl#L92-L102

OTOH, there's no pedant test for that, and it's an existing bug, and I probably won't have time to fix all that. So my inclination is to leave it as is until the problem comes up again.